go/kbfs/libfs/httprootfs.go

```diff
@@ -0,0 +1,52 @@
+package libfs
+
+import (
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	billy "gopkg.in/src-d/go-billy.v4"
+)
+
+type httpRootFileSystem struct {
+	rfs *RootFS
+}
+
+var _ http.FileSystem = httpRootFileSystem{}
+
+type fileOnly struct {
+	billy.File
+	rfs *RootFS
+}
+
+// Readdir implements the http.File interface.
+func (fo fileOnly) Readdir(count int) ([]os.FileInfo, error) {
+	return nil, errors.New("not supported")
+}
+
+// Stat implements the http.File interface.
+func (fo fileOnly) Stat() (os.FileInfo, error) {
+	return fo.rfs.Stat(fo.File.Name())
+}
+
+func (hrfs httpRootFileSystem) Open(filename string) (entry http.File, err error) {
+	hrfs.rfs.log.CDebugf(hrfs.rfs.ctx, "hfs.Open %s", filename)
```

Comment:
I think we can do without this logging here, and then you can get rid of the new `ctx` in `RootFS`.  I would probably just need to re-do it anyway for my obfuscation PR, and I don't think it adds much.  But if you really want it for some reason, we can leave it.